### ISSUE TITLE
Request iranian domain list

### ISSUE THREAD

**Issue by debMan** [v2fly/domain-list-community#1337]
Can anybody help and make a list obtained from the official regulations of Iran?

https://eservices.ito.gov.ir/Page/IPList

**Comment by rootmelo92118** [comment]
Could you please upload this list here? It looks like reject the connection outside Iran.

**Comment by IceCodeNew** [comment]
> Can anybody help and make a list obtained from the official regulations of Iran?
> 
> [eservices.ito.gov.ir/Page/IPList](https://eservices.ito.gov.ir/Page/IPList)

Would you please have a look at #1294?
If there are domains that were missing in this PR, you can submit another PR for amendment.

**Comment by debMan** [comment]
Dear @IceCodeNew @rootmelo92118, you both are correct. The domain is accessible only inside Iran.
It contains a table with paginations, so I have to crawl them and add them as text.
![image](https://user-images.githubusercontent.com/12995147/205301723-f0a1bacc-1bb0-4918-9a79-afcbb37d066c.png)

> Would you please have a look at #1294? If there are domains that were missing in this PR, you can submit another PR for amendment.
> 
The PR is correct to some extent, but there are more domains which are not been added yet.

**Comment by IceCodeNew** [comment]
> > Would you please have a look at #1294? If there are domains that were missing in this PR, you can submit another PR for amendment.
> 
> The PR is correct to some extent, but there are more domains which are not been added yet.

It would be nice if you could submit a PR for the missing domains since I and @rootmelo92118 can only make the PR based on a screenshot, which is pretty inconvenient.

**Comment by rootmelo92118** [comment]
@debMan Is it?
[Export-14010911183802.xls](https://github.com/v2fly/domain-list-community/files/10142439/Export-14010911183802.xls)

**Comment by debMan** [comment]
> > > Would you please have a look at #1294? If there are domains that were missing in this PR, you can submit another PR for amendment.
> > 
> > 
> > The PR is correct to some extent, but there are more domains which are not been added yet.
> 
> It would be nice if you could submit a PR for the missing domains since I and @rootmelo92118 can only make the PR based on a screenshot, which is pretty inconvenient.

Sure, thanks. I'll work on it ASAP.

**Comment by debMan** [comment]
> @debMan Is it? [Export-14010911183802.xls](https://github.com/v2fly/domain-list-community/files/10142439/Export-14010911183802.xls)

Does `geopip` of Iran covers all iranian non `.ir` domains?

**Comment by rootmelo92118** [comment]
> > @debMan Is it? [Export-14010911183802.xls](https://github.com/v2fly/domain-list-community/files/10142439/Export-14010911183802.xls)
> 
> Does `geopip` of Iran covers all iranian non `.ir` domains?

I am not really sure.

**Comment by rootmelo92118** [comment]
> > @debMan Is it? [Export-14010911183802.xls](https://github.com/v2fly/domain-list-community/files/10142439/Export-14010911183802.xls)
> 
> I have an idea about that. more than a half of these domains end with `.ir` (Iran domain). so I thinks it's better to delete them and use a regexp : `regexp:.*\.ir$`. I use this and geoip of Iran. It works fine.

I don't think that is a good idea. If we use `regexp:.+\.ir$`, it will make all `.ir` domains go through proxy or direct.
And you can use `ir` in the rules. That is equal to `regexp:.+\.ir$`

**Comment by rootmelo92118** [comment]
> > @debMan Is it? [Export-14010911183802.xls](https://github.com/v2fly/domain-list-community/files/10142439/Export-14010911183802.xls)
> 
> Does `geopip` of Iran covers all iranian non `.ir` domains?

https://github.com/v2fly/geoip

**Comment by mmrabbani** [comment]
@debMan 
category-ir is a list of domains that block access from outside of iran. it is a very short list
My pull request (ir) contains all domains hosted in iran or domain that will not be blocked by iran. It is a large list. I think accepting my pull request may answer your need.
[https://github.com/v2fly/domain-list-community/pull/1375](https://github.com/v2fly/domain-list-community/pull/1375)

**Comment by MJamshidnejad** [comment]
> > > @debMan Is it? [Export-14010911183802.xls](https://github.com/v2fly/domain-list-community/files/10142439/Export-14010911183802.xls)
> > 
> > 
> > I have an idea about that. more than a half of these domains end with `.ir` (Iran domain). so I thinks it's better to delete them and use a regexp : `regexp:.*\.ir$`. I use this and geoip of Iran. It works fine.
> 
> I don't think that is a good idea. If we use `regexp:.+\.ir$`, it will make all `.ir` domains go through proxy or direct outbound. By the way, you can use `ir` in the rules. That is equal to `regexp:.+\.ir$`

There's no difference in practice between them. Because all domain have at least a letter before dot. I use it with nekoray 2.6 and v2fly 5.1.0.

**Comment by bootmortis** [comment]
I like to mention these two comments here:

https://github.com/v2fly/domain-list-community/pull/1438#issuecomment-1428404334

https://github.com/v2fly/domain-list-community/pull/1438#issuecomment-1428484966


IMO, this issue and related PRs could be closed, as the ITO gov is not a reliable source.
